### PR TITLE
Complete coverage of iot_mqtt_subscription.c

### PR DESCRIPTION
**Pull request by gordonwang0**

### CODE REVIEW

File: libraries/standard/mqtt/src/iot_mqtt_validate.c [hunk 1]
```diff
@@ -354,41 +354,41 @@ static bool _validateSubscription( bool awsIotMqttMode,
     /* Check for a valid QoS and callback function when subscribing. */
     if( operation == IOT_MQTT_SUBSCRIBE )
     {
-        status = _validateQos( pSubscription->qos );
+        if( pSubscription->callback.function == NULL )
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/src/iot_mqtt_validate.c]: Due to this change, line 364 could be changed to an `else`

File: libraries/standard/mqtt/src/iot_mqtt_subscription.c [hunk 2]
```diff
@@ -161,7 +161,7 @@ static bool _matchEndWildcards( const char * pTopicFilter,
     if( endChar == true )
     {
         /* Determine if the topic filter ends with the '#' wildcard. */
-        status = ( pTopicFilter[ filterIndex + 1U ] == '/' ) && ( pTopicFilter[ filterIndex + 2U ] == '#' );
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/src/iot_mqtt_subscription.c]: Is `pTopicFilter[filterIndex+1] == '/'` not checked due to being validated elsewhere?

File: libraries/standard/mqtt/src/iot_mqtt_validate.c (hunk 1 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/src/iot_mqtt_validate.c]: changed

File: libraries/standard/mqtt/src/iot_mqtt_subscription.c (hunk 2 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/src/iot_mqtt_subscription.c]: Correct

**Pull request merged**